sdk/eventhub/azure-eventhub/azure/eventhub/_transport/_pyamqp_transport.py

```diff
@@ -532,8 +550,11 @@ def _create_eventhub_exception(exception):
         if isinstance(exception, errors.AuthenticationException):
             error = AuthenticationError(str(exception), exception)
         elif isinstance(exception, errors.AMQPLinkError):
-            error = ConnectError(str(exception), exception)
-        # TODO: do we need MessageHanlderError in amqp any more
+            if exception.condition in PyamqpTransport.ERROR_CONDITIONS:
+                error = ConnectionLostError(str(exception), exception)
+            else:
+                error = ConnectError(str(exception), exception)
```

Comment:
exception parity with uampq

Reply:
It looks like we're using ConnectionLostError for effectively all the protocol error conditions - what are the conditions we would expect to raise ConnectError?

Reply:
We raise the ConnectError when the error condition starts with "com." (aka. the "VendorLinkDetach" from uamqp). 

https://github.com/Azure/azure-sdk-for-python/blob/b2d5a9a2d0d70d400da562ecef5e5dafa14cc729/sdk/eventhub/azure-eventhub/azure/eventhub/_transport/_uamqp_transport.py#L575

Adding this comment to the code here.